File: app/src/main/java/net/orner/dollarcribbage/app/MainActivity.java
package net.orner.dollarcribbage.app;

import android.app.Activity;
import android.location.Criteria;
import android.os.Bundle;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.TextView;
import android.widget.Toast;

import java.util.TooManyListenersException;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }


    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();
        if (id == R.id.action_settings) {
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    public void onClick(View view) {
        TextView serialView = (TextView) findViewById(R.id.serialnumber);
        String handstring = serialView.getText().toString();

        CribbageCalc cribbageCalc = new CribbageCalc();
        if (!cribbageCalc.isValidHand(handstring))
        {
            Toast.makeText(this, "Invalid Serial Number", Toast.LENGTH_LONG).show();
            return;
        }

        int score = cribbageCalc.CalcHand(handstring);

        TextView resultsView = (TextView) findViewById(R.id.textResults);

        String results = "Score " + score;


        resultsView.setText(results);

    }

}

File: app/src/main/java/net/orner/dollarcribbage/app/MainActivity.kt
package net.orner.dollarcribbage.app

import android.app.Activity
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.TextView
import android.widget.Toast

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }


    override fun onCreateOptionsMenu(menu: Menu): Boolean {

        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        val id = item.itemId
        return if (id == R.id.action_settings) {
            true
        } else super.onOptionsItemSelected(item)
    }

    fun onClick(view: View) {
        val serialView = findViewById<View>(R.id.serialnumber) as TextView
        val handString = serialView.text.toString()

        val cribbageCalc = CribbageCalc()
        if (!cribbageCalc.isValidHand(handString)) {
            Toast.makeText(this, "Invalid Serial Number", Toast.LENGTH_LONG).show()
            return
        }

        val score = cribbageCalc.calcHand(handString)

        val resultsView = findViewById<View>(R.id.textResults) as TextView

        val results = "Score $score"


        resultsView.text = results

    }

}
